src/NHibernate/Cache/ReadWriteCache.cs

```diff
@@ -24,11 +41,20 @@ namespace NHibernate.Cache
 	/// </remarks>
 	public partial class ReadWriteCache : IBatchableCacheConcurrencyStrategy
 	{
+		//TODO 6.0: Merge with ILockable
+		internal interface ILockableNextVer
```

Comment:
Maybe `IMinimalPutAwareLockable`? I don't like V2, nextVer, etc names. They are too broad. Also, maybe make this an abstract class?

Reply:
We've already discussed it. It's bad for open API. It's fine for temp internal interface and requires less frictions if additional changes are required for old interface. But OK it's really nit for me in this context let it be `IMinimalPutAwareLockable`.